src/main/java/de/tum/in/www1/artemis/web/rest/AthenaResource.java

```diff
@@ -1,81 +1,199 @@
 
     private final TextSubmissionRepository textSubmissionRepository;
 
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final ProgrammingSubmissionRepository programmingSubmissionRepository;
+
     private final AuthorizationCheckService authCheckService;
 
     private final AthenaFeedbackSuggestionsService athenaFeedbackSuggestionsService;
 
+    private final AthenaRepositoryExportService athenaRepositoryExportService;
+
     /**
      * The AthenaResource provides an endpoint for the client to fetch feedback suggestions from Athena.
      */
-    public AthenaResource(AthenaFeedbackSuggestionsService athenaFeedbackSuggestionsService, TextExerciseRepository textExerciseRepository,
-            TextSubmissionRepository textSubmissionRepository, AuthorizationCheckService authCheckService) {
-        this.athenaFeedbackSuggestionsService = athenaFeedbackSuggestionsService;
+    public AthenaResource(TextExerciseRepository textExerciseRepository, TextSubmissionRepository textSubmissionRepository,
+            ProgrammingExerciseRepository programmingExerciseRepository, ProgrammingSubmissionRepository programmingSubmissionRepository,
+            AuthorizationCheckService authCheckService, AthenaFeedbackSuggestionsService athenaFeedbackSuggestionsService,
+            AthenaRepositoryExportService athenaRepositoryExportService) {
         this.textExerciseRepository = textExerciseRepository;
         this.textSubmissionRepository = textSubmissionRepository;
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.programmingSubmissionRepository = programmingSubmissionRepository;
         this.authCheckService = authCheckService;
+        this.athenaFeedbackSuggestionsService = athenaFeedbackSuggestionsService;
+        this.athenaRepositoryExportService = athenaRepositoryExportService;
     }
 
     /**
-     * GET athena/exercises/:exerciseId/submissions/:submissionId/feedback-suggestions : Get feedback suggestions from Athena
+     * GET athena/text-exercises/:exerciseId/submissions/:submissionId/feedback-suggestions : Get feedback suggestions from Athena for a text exercise
      *
      * @param exerciseId   the id of the exercise the submission belongs to
      * @param submissionId the id of the submission to get feedback suggestions for
      * @return 200 Ok if successful with the corresponding result as body
      */
-    @GetMapping("exercises/{exerciseId}/submissions/{submissionId}/feedback-suggestions")
+    @GetMapping("athena/text-exercises/{exerciseId}/submissions/{submissionId}/feedback-suggestions")
     @EnforceAtLeastTutor
-    public ResponseEntity<List<TextBlockRef>> getFeedbackSuggestions(@PathVariable long exerciseId, @PathVariable long submissionId) {
+    public ResponseEntity<List<TextFeedbackDTO>> getTextFeedbackSuggestions(@PathVariable long exerciseId, @PathVariable long submissionId) {
```

Comment:
Comment for reviewers: As far as I know there is no elegant way to have either `TextFeedbackDTO` or `ProgrammingFeedbackDTO` as a result without ugly hacks, so I created two completely separate endpoints here.